src/hyperv/hyperv_network_api_v1.c

```diff
@@ -71,104 +71,142 @@ hyperv1MsvmVirtualSwitchToNetwork(virConnectPtr conn,
     return result;
 }
 
-/* exported API functions */
-
-int
-hyperv1ConnectListNetworks(virConnectPtr conn, char **const names, int maxnames)
+static int
+hyperv1GetVirtualSwitchList(hypervPrivate *priv,
+                            const char *filter,
```

Comment:
You don't clean up this filter string anywhere. It should probably be cleaned up in this function.